apps/web/src/components/ClientAnalyticsScript/ClientAnalyticsScript.tsx

```diff
@@ -1,22 +1,62 @@
-import { useCallback } from 'react';
+'use client';
+
+import { useCallback, useEffect, useMemo, useRef } from 'react';
 import { useSavedTrackingPreference, useCookie } from '@coinbase/cookie-manager';
-import { useRouter } from 'next/router';
+import { usePathname } from 'next/navigation';
+
 import Script from 'next/script';
 import initCCA from '../../utils/initCCA';
 
+export type NextJsRouterEventTypes =
+  | 'routeChangeStart'
+  | 'beforeHistoryChange'
+  | 'routeChangeComplete'
+  | 'routeChangeError'
+  | 'hashChangeStart'
+  | 'hashChangeComplete';
+
+export type AnalyticsEventHandler = () => void;
+
+export type NextJSRouter = Record<string, unknown> & {
+  events: {
+    on: (event: NextJsRouterEventTypes, callback: AnalyticsEventHandler) => void;
+  };
+};
+
 export default function ClientAnalyticsScript() {
-  const router = useRouter();
   const trackingPreference = useSavedTrackingPreference();
   // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
   const [deviceIdCookie, setDeviceIdCookie] = useCookie('base_device_id');
 
+  const pathname = usePathname();
+  const handler = useRef<AnalyticsEventHandler | null>(null);
+
+  // Mock the old router since our CCA doesn't support Next App router yet
+  const oldRouterEvent: NextJSRouter = useMemo(() => {
+    return {
+      events: {
+        on: (event: NextJsRouterEventTypes, analyticsHandler: AnalyticsEventHandler) => {
+          if (analyticsHandler) {
+            handler.current = analyticsHandler;
+          }
+        },
+      },
+    };
+  }, []);
+
+  useEffect(() => {
```

Comment:
Let's just explain/comment this a bit more since it feels like a bit of a hack, although quite an elegant one